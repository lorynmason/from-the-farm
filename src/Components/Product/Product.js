import React, { Component } from 'react';

class Product extends Component {

  render() {
    let { name, price, unit, description, vendorName } = this.props.product;
    const newPrice = price/100;
    price = `$${newPrice.toFixed(2)}`;
    return (
      <article className="product">
        <h2>{name}</h2>
        <p>{vendorName}</p>
        <p>{description}</p>
        <div className="pricing-info">
          <p>{price}/{unit}</p>
        </div>
      </article>
    )
  }
}

export default Product;