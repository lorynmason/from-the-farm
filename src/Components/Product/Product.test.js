import React from 'react';
import Product from './Product';
import { shallow } from 'enzyme'

describe('Product', () => {
  it('should match snapshot', () => {
    let wrapper = shallow(<Product product={{
      name: 'chile pepper',
      description: 'Pueblo hot roasted',
      price: 20.00,
      quantity: 1,
      unit: 'bushel'
    }} />)
    expect(wrapper).toMatchSnapshot()
  });
})
