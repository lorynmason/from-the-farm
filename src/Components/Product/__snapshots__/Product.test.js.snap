// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Product
    product={
      Object {
        "description": "Pueblo hot roasted",
        "name": "chile pepper",
        "price": 20,
        "quantity": 1,
        "unit": "bushel",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          chile pepper
        </h2>,
        <p />,
        <p>
          Pueblo hot roasted
        </p>,
        <div
          className="pricing-info"
        >
          <p>
            $0.20
            /
            bushel
          </p>
        </div>,
      ],
      "className": "product",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "chile pepper",
        },
        "ref": null,
        "rendered": "chile pepper",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Pueblo hot roasted",
        },
        "ref": null,
        "rendered": "Pueblo hot roasted",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <p>
            $0.20
            /
            bushel
          </p>,
          "className": "pricing-info",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "$0.20",
              "/",
              "bushel",
            ],
          },
          "ref": null,
          "rendered": Array [
            "$0.20",
            "/",
            "bushel",
          ],
          "type": "p",
        },
        "type": "div",
      },
    ],
    "type": "article",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            chile pepper
          </h2>,
          <p />,
          <p>
            Pueblo hot roasted
          </p>,
          <div
            className="pricing-info"
          >
            <p>
              $0.20
              /
              bushel
            </p>
          </div>,
        ],
        "className": "product",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "chile pepper",
          },
          "ref": null,
          "rendered": "chile pepper",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Pueblo hot roasted",
          },
          "ref": null,
          "rendered": "Pueblo hot roasted",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              $0.20
              /
              bushel
            </p>,
            "className": "pricing-info",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "$0.20",
                "/",
                "bushel",
              ],
            },
            "ref": null,
            "rendered": Array [
              "$0.20",
              "/",
              "bushel",
            ],
            "type": "p",
          },
          "type": "div",
        },
      ],
      "type": "article",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
